client/src/App.jsx

```diff
@@ -1,7 +1,119 @@
-import './App.css';
+import { useState, useEffect } from 'react';
+import EntryList from './EntryList';
+import PageHeader from './PageHeader';
+import EntryForm from './EntryForm';
+import Modal from './Modal';
+import data from './data';
 
 function App() {
-  return <></>;
+  //NOTE: When the App is first load, it will check if there are any data in the local storage that can be used. If there are no data that can be used, it will use the given empty data as the initial value. Once the page is render for the first time, the callback function inside of the useEffect will be called and set the empty data in the local storage. Then, any changes to the savedData after clicking save on the form will be updated and save to the local storage. Determine the initialData is also needed to load the saved entries after page refresh.
+
+  const initialDataResult = initialData();
+  const [savedData, setSavedData] = useState(initialDataResult);
+  const [view, setView] = useState('entries');
+  const [entryTitle, setEntryTitle] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [notes, setNotes] = useState('');
+  const PageDisplay = viewPageDisplay();
+
+  useEffect(() => {
+    const dataJSON = JSON.stringify(savedData);
+    localStorage.setItem('code-journal-data', dataJSON);
+  }, [savedData]);
+
+  function initialData() {
+    const localData = JSON.parse(localStorage.getItem('code-journal-data'));
+
+    if (!localData) {
+      return data;
+    }
+    return localData;
+  }
+
+  function viewPageDisplay() {
+    if (view === 'entry-form') {
+      return (
+        <EntryForm
+          formTitle="New Entry"
+          formValue={[entryTitle, imgUrl, notes]}
+          onSubmit={handleFormSubmit}
+          onChange={handleFormChange}
+        />
+      );
+    }
+
+    if (view === 'edit-form') {
+      return (
+        <EntryForm
+          formTitle="Edit Entry"
+          formValue={[entryTitle, imgUrl, notes]}
+          onSubmit={handleFormSubmit}
+          onChange={handleFormChange}
+        />
+      );
+    }
+
+    return <EntryList data={savedData} onClick={handleViewForm} />;
```

Comment:
`onClick` is a tad confusing as a prop name in the child components. I would literally keep the name `handleViewForm = {handleViewForm}`

Reply:
🚨 HOWEVER: I would just pass `handleView = {handleView}`  (or `handleViewForm = {handleViewForm}` if really needed) and add an `onClick` handler to the component/button/link you click on! See the example in `EntryList` and below

```
function App {

  function handleView(page) {
    setView(page);
  }
  
<EntryList handleView={handleView}> 
}
```

```
function EntryList({ handleView }) {

return <button onClick={() => handleView('insert your view here')}
} 
```
This will make changing your view easier!!